public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -245,6 +238,25 @@ export class ElasticDatasource
     return this.request('POST', url, data, { 'Content-Type': 'application/x-ndjson' });
   }
 
+  private async isEs7Cached(): Promise<boolean> {
+    const data = await lastValueFrom(this.request('GET', '/'));
+    const versionNumber = data?.version?.number;
+    if (typeof versionNumber !== 'string') {
+      return false; // default value
+    }
+    return versionNumber.startsWith('7.');
+  }
+
+  async isEs7(): Promise<boolean> {
```

Comment:
What is the benefit of keeping `isEs7` vs keeping the actual version?

Reply:
the general problem is that we cannot always generate an actual version. what we can store is a string that is "actual version, or some other string".
so every time code calls datasource.getDbVersion(), they have to understand, that they may have an invalid-string in their hands, and they always have to call `semver.valid()` first, before doing anything. that's why i'm looking for some other way to handle this, where i can wrap this "problem" inside the datasource.

but maybe there's some other way, not sure.

Reply:
anyway, i adjusted the code, now the datasource returns the actual db-version.